src/rules/__tests__/no-deprecated-functions.test.ts

```diff
@@ -154,6 +155,20 @@ describe('the rule', () => {
       expect(() => {
         const linter = new TSESLint.Linter();
 
+        /* istanbul ignore if */
+        if (usingFlatConfig()) {
```

Comment:
why is this needed?

Reply:
Because we're effectively running ESLint here per usual - so when using ESLint v9 we have to use flatconfig and when using pre-v9 we have to use legacy config.

While `linter.verify` supports being passed both types of configs, since we have call `defineRule` in pre-v9 (which errors in v9) this felt the cleanest way to address it (though either way we'll always need `usingFlatConfig`.